src/lib.rs

```diff
@@ -3287,32 +3286,39 @@ impl Build {
                     self.cmd(&name)
                 } else if self.get_is_cross_compile()? {
                     match self.prefix_for_target(&self.get_raw_target()?) {
-                        Some(p) => {
+                        Some(prefix) => {
                             // GCC uses $target-gcc-ar, whereas binutils uses $target-ar -- try both.
                             // Prefer -ar if it exists, as builds of `-gcc-ar` have been observed to be
                             // outright broken (such as when targeting freebsd with `--disable-lto`
                             // toolchain where the archiver attempts to load the LTO plugin anyway but
                             // fails to find one).
                             //
                             // The same applies to ranlib.
-                            let mut chosen = default;
-                            for &infix in &["", "-gcc"] {
-                                let target_p = format!("{}{}-{}", p, infix, tool);
-                                if Command::new(&target_p).output().is_ok() {
-                                    chosen = target_p;
-                                    break;
-                                }
-                            }
+                            let chosen = ["", "-gcc"]
+                                .iter()
+                                .filter_map(|infix| {
+                                    let target_p = format!("{prefix}{infix}-{tool}");
+                                    let status = Command::new(&target_p)
+                                        .arg("--version")
+                                        .stdin(Stdio::null())
+                                        .stdout(Stdio::null())
+                                        .stderr(Stdio::null())
+                                        .status()
```

Comment:
Is there a reason why we don't invoke `which $target_p` instead? It seems like that would have the same effect of determining if the command exists (and it would be easier to optimize/understand, since we could implement the `which` algorithm ourselves).

Reply:
Which also has quite a few deps unfortunately. The core logic would be re-implementable in `cc` though.

Reply:
> Which also has quite a few deps unfortunately. The core logic would be re-implementable in `cc` though.

Just slightly worried about having more code to be maintained.